src/resmom/requests.c

```diff
@@ -765,15 +783,37 @@ message_job(job *pjob, enum job_file jft, char *text)
 		pstr[len++] = '\n';	/* append new-line */
 		text = pstr;
 	}
-	(void)write(fds, text, len);
 #ifdef	WIN32
+	(void)write(fds, text, len);
 	(void)_commit(fds);
+#else
+	flags = fcntl(fds, F_GETFL, 0);
+	fcntl(fds, F_SETFL, flags | O_NONBLOCK);
+
+	while ((slept < 5) && (total_bytes_written < len)) {
+		bytes_written = write(fds, text, len-total_bytes_written);
+		total_bytes_written += bytes_written;
+		text += bytes_written;
+		if (bytes_written <= 0) {
+			if (errno == EAGAIN || errno == EWOULDBLOCK)
+			{
+				sleep(1);
+				slept++;
+			} else {
+					(void)close(fds);
+					return PBSE_MOMREJECT;
+			}
+		}
+	}
 #endif
 	(void)close(fds);
 	if (pstr)
 		free(pstr);
 
-	return PBSE_NONE;
+	if (total_bytes_written == len)
```

Comment:
What if above write() call returns -1 (in case errno=EAGAIN) at least one time in the loop ? Does this condition satisfy on next successful write?

Reply:
yes.